### ISSUE TITLE
AVT 1 - Carbon X Structured List Component DAP violation

### ISSUE THREAD

**Issue by snidersd** [carbon-design-system/carbon#1882]
## Environment
macOS Mojave version 10.14.2
Chrome version 72.0.3626.96 (Official Build) (64-bit)

## Detailed description
AVT 1 Run DAP on the example and there are 4 violations. See screenshot.
<img width="1270" alt="structurelist" src="https://user-images.githubusercontent.com/21676914/53202207-87984200-35eb-11e9-893a-d18ba3b508f9.png">

**Comment by snidersd** [comment]
This is still an open issue and it applies to React as well. See screenshot below:
<img width="1322" alt="Screen Shot 2019-06-12 at 1 57 57 PM" src="https://user-images.githubusercontent.com/21676914/59379344-fd9bf280-8d1b-11e9-9899-9834031e5c67.png">

**Comment by joshblack** [comment]
Resolution for this is to drop `role="presentation"` from `<label>`. This should address all DAP violations.

**Comment by emyarod** [comment]
is the `for` attribute target violation just a false positive then?

**Comment by joshblack** [comment]
@emyarod yeah, I think it's messing up because of the role usage but could be wrong. Did some quick testing when going through things and it seemed like dropping the role would also address the for violation, but definitely not sure.

**Comment by joshblack** [comment]
Tried to solve this today, it seems like the underlying markup is wrong for the use-case. Updating that should also include addressing this violation. The example we would most likely reference is: https://www.w3.org/TR/wai-aria-practices/examples/grid/advancedDataGrid.html but it seems like it has not been finished yet.

Looks like we'll need to take a look at the following for this component:

- Update the markup either to table or grid
  - Can we use table markup?
  - Can we use ARIA for table if not?
  - Does grid make sense here for interactions?
- Figure out how to do row-level interactions like selection for this use-case

**Comment by emyarod** [comment]
> row-level interactions like selection

might be related to/briefly explored this for #2820
